### PR TITLE
Update error status code documentation a little bit.

### LINKED ISSUE #2174: Document API HTTP status codes in api-conventions.md and file bugs for inconsistencies in API implementation
Status and error behavior is not well documented and @erictune has found it to be inconsistent, when writing auth_test.go. We need consistent behavior with respect to HTTP status codes and API errors so that clients can respond in a uniform manner regardless of the object kind, including API plugins.

One source to start with is:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/errors/errors.go

/cc @smarterclayton @lavalamp

### PR DESCRIPTION
**Pull request by brendandburns**
@bgrant0607 

Partially addresses #2174

### CODE REVIEW

File: docs/api-conventions.md [hunk 1]
```diff
@@ -238,8 +238,6 @@ The following HTTP status codes may be returned by the API.
 
 * `200 StatusOK`
   * Indicates that the request completed successfully.
-* `201 StatusCreated`
```
Comment (bgrant0607) [inline docs/api-conventions.md]: Why are you removing 201?

We do return 201:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/test/integration/auth_test.go#L201

#2031

File: docs/api-conventions.md [hunk 2]
```diff
@@ -297,11 +295,10 @@ The following HTTP status codes may be returned by the API.
...
 
-A ```Status``` kind MAY be returned by an API when an operation is successful (i.e. when an HTTP 200 status code is returned). In particular, delete APIs return the ```Status``` kind. The success status object simply contains a ```Status``` field set to ```Successs```.
-
-A ```Status``` kind SHOULD be returned by an API when an operation is not successful (i.e. when the server would return a non 2xx HTTP status code). The status object contains fields for humans and machine consumers of the API to get more detailed information for the cause of the failure. The information in the status object supplements, but does not override, the HTTP status code's meaning.
+A ```Status``` kind will be returned by an API when an operation is not successful (i.e. when the server would return a non 2xx HTTP status code). The status object is encoded as JSON and provided as the body of the response.  The status object contains fields for humans and machine consumers of the API to get more detailed information for the cause of the failure. The information in the status object supplements, but does not override, the HTTP status code's meaning.
```
Comment (bgrant0607) [inline docs/api-conventions.md]: We also currently return Status on delete.

**Comment by bgrant0607** [comment]
Also, this doesn't close the issue. auth_test might be reasonably comprehensive, but we're not yet documenting status codes in the swagger, which we could.

**Comment by brendandburns** [comment]
Ok, I thought with the removal of operations, we no longer returned 201.
I'll re-add it, and update the section on status to include delete.

Thanks
Brendan

File: docs/api-conventions.md (hunk 1 above)
Comment (brendandburns) [inline docs/api-conventions.md]: fixed.

File: docs/api-conventions.md (hunk 2 above)
Comment (brendandburns) [inline docs/api-conventions.md]: done.

**Comment by brendandburns** [comment]
Comments addressed.

PTAL.
Thanks!
--brendan

**Comment by bgrant0607** [comment]
LGTM

**Pull request merged**